extensions/rancherleader/rancherleader.go

```diff
@@ -0,0 +1,40 @@
+package rancherleader
+
+import (
+	"encoding/json"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/extensions/configmaps"
+)
+
+const (
+	KubeSystemNamespace     = "kube-system"
+	RancherConfigMap        = "cattle-controllers"
+	RancherLeaderAnnotation = "control-plane.alpha.kubernetes.io/leader"
+)
+
+// GetRancherLeaderPod is a helper function to retrieve the name of the rancher leader pod
+func GetRancherLeaderPod(client *rancher.Client) (string, error) {
```

Comment:
Great addition, just a small naming nit here;

```suggestion
func GetRancherLeaderPodName(client *rancher.Client) (string, error) {
```